File: task-03/SubTask3/diamond_pattern.cpp
#include <iostream>
#include <iomanip>

int main() {
    int n;
    std::cout << "Enter a number: ";
    std::cin >> n;

    for (int i = 0; i < n; i++) {
        std::cout << std::setw(n - i) << "";
        for (int j = 0; j < 2 * i + 1; j++) {
            std::cout << "*";
        }
        std::cout << std::endl;
    }
    for (int i = n - 2; i >= 0; i--) {
        std::cout << std::setw(n - i) << "";
        for (int j = 0; j < 2 * i + 1; j++) {
            std::cout << "*";
        }
        std::cout << std::endl;
    }
    return 0;
}

File: task-03/SubTask3/diamond_pattern.hs
main :: IO ()
main = do
    putStrLn "Enter a number: "
    n <- readLn
    mapM_ putStrLn (diamond n)

diamond :: Int -> [String]
diamond n = [replicate (n-i-1) ' ' ++ replicate (2*i+1) '*' | i <- [0..n-1]]
            ++ [replicate (n-i-1) ' ' ++ replicate (2*i+1) '*' | i <- [n-2,n-3..0]]
